### PR TITLE
More metrics

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

File: log/sequencer.go [hunk 1]
```diff
@@ -19,17 +19,54 @@ package log
...
+	seqInitTreeLatency     monitoring.Histogram
+	seqWriteTreeLatency    monitoring.Histogram
+	seqUpdateLeavesLatency monitoring.Histogram
+	seqSetNodesLatency     monitoring.Histogram
+	seqStoreRootLatency    monitoring.Histogram
+	seqCommitLatency       monitoring.Histogram
+)
+
+func createMetrics(mf monitoring.MetricFactory) {
+	if mf == nil {
```
Comment (Martin2112) [inline log/sequencer.go]: This is defensive? Could we require a non nil metrics factory and move the problem up or is that no better?

File: storage/mysql/log_storage.go [hunk 2]
```diff
@@ -325,6 +347,9 @@ func (t *logTreeTX) DequeueLeaves(ctx context.Context, limit int, cutoffTime tim
 	if rows.Err() != nil {
 		return nil, rows.Err()
 	}
+	label := labelForTX(t)
+	selectDuration := time.Now().Sub(start)
+	dequeueSelectLatency.Observe(float64(selectDuration/time.Millisecond), label)
```
Comment (Martin2112) [inline storage/mysql/log_storage.go]: Would a local observeAsMillis(metric, time.Duration) func help to remove the boilerplate division?

File: log/sequencer.go (hunk 1 above)
Comment (daviddrysdale) [inline log/sequencer.go]: It's basically defensive against having to plumb a metrics factory through every bit of test code that uses a sequencer.

File: storage/mysql/log_storage.go (hunk 2 above)
Comment (daviddrysdale) [inline storage/mysql/log_storage.go]: Good point, done.

File: log/sequencer.go (hunk 1 above)
Comment (Martin2112) [inline log/sequencer.go]: OK, that sounds reasonable.

**Pull request merged**